lnrpc/rpc.proto

```diff
@@ -3098,3 +3098,163 @@ message BakeMacaroonResponse {
     /// The hex encoded macaroon, serialized in binary format.
     string macaroon = 1 [json_name = "macaroon"];
 }
+
+
+message Failure {
```

Comment:
So this is not moved to `lnrpc` because of `SendToRoute`. At least not yet. For this PR it is done to be able to expose the raw failure in `ListPayments`. I think the options that we have are:
* Not expose the raw failure
* Copy/paste the proto message...
* Maybe duplicate `ListPayments` in `routerrpc`?

Thoughts?

Reply:
Discussed offline and decided to keep change as it is. `routerrpc` is experimental and this change is only breaking during building against the proto.

Reply:
We could also easily split the RPC changes into its own PR, since they are not strictly necessary for this PR. If we need more time to figure out the best way to solve this.

Reply:
Non-blocking, but I'd prefer to just copy over the protos compared to breaking users of the existing `routerrpc` package (some wallets now use this exclusively).